### ISSUE TITLE
WinGet for Linux

### ISSUE THREAD

**Issue by BurningPho3nix** [microsoft/winget-cli#3889]
### Description of the new feature / enhancement

My idea would be if you could make a functioning version of winget for Linux

### Proposed technical implementation details

Please don't take this serious, although it would be funny

**Comment by vedantmgoyal2009** [comment]
you already have Snap, apt, and brew for Linux. Winget is just like those Linux package managers but for Windows. The name itself  suggests that.

**Comment by willyamcts** [comment]
I agree @vedantmgoyal2009.

One option would be to for Linux can do  query in WinGet repository, logically not upgrade nothing, only query and search.

If the idea is valid

**Comment by danlor-dev** [comment]
What a great Idea, one Package Source and deploy it on multiple Clients. Real Use Case for me is, deploy for example Visual Studio Code in Package Repository and i could it deploy on Windows, Linux a.s.o.. Create it one time and deploy it everywhere ;-).  

For now i have to create a package for winget, a package for snap or others. Multiple maintenance for each target platform :-( Not so great idea.

**Comment by beedell-roke** [comment]
> https://github.com/microsoft/winget-cli/issues/3889#issue-1992743601

I see a genuine use case for this. Linux-based OSes tend to have infrastructure associated with them that provides compiled packages. However, where that isn't available, users are expected to understand the FHS, and thereby extract self-contained archives into `/opt`, etcetera. The sole alternatives are `flatpak` and `snap`, but they simultaneously introduce significant security abstraction that renders even basic DBus communication and filesystem manipulation difficult. [^2]

[^2]: [`flathub/com.google.AndroidStudio/issues/268`](https://github.com/flathub/com.google.AndroidStudio/issues/268#issuecomment-2832400350)

Consequently, WinGet, which functions significantly more similarly to something like Obtanium (albeit with a centralised package *manifest* repository, and support for unpackaged software), would allow users to install software that isn't deliberately packaged by upstream or downstream for a specific distribution model. Instead, they merely request that a manifest be generated for whatever URI they provide to the archive file, then download that through WinGet when added.

`brew` is probably the closest equivalent, but since it itself isn't installable as a native package in any popular distribution I've tried, it can be dangerous to the OS's integrity. [^1]

[^1]: [`discussion.fedoraproject.org/t/36013/38`](https://discussion.fedoraproject.org/t/install-and-use-homebrew-on-fedora/36013/38?u=rokejulianlockhart#:~:text=github.com/orgs/Homebrew/discussions/6033,discussion.fedoraproject.org/t/36013/34%20%E2%86%A9%EF%B8%8E)

#### Tangential

<p></p><details><p></p>

Azure Linux is a controlled environment. However, it is the basis for WSL. If ever Microsoft wants a first-party WSL image, this would be the most significant step they could make toward it (bar a Wayland display server).

<p></p></details><p></p>

**Comment by aristosh** [comment]
How about the installed app will be sandboxed by wine? No point but sounds cool 

> I agree [@vedantmgoyal2009](https://github.com/vedantmgoyal2009).
> 
> One option would be to for Linux can do query in WinGet repository, logically not upgrade nothing, only query and search.
> 
> If the idea is valid

How about if the installed app will be sandboxed inside Wine? No point but sounds cool 😄

**Comment by beedell-roke** [comment]
@aristosh, no. It'd be a lot more feasible if there were merely new additions to the current manifest standard and a Linux-compatible client. Wine barely supports any modern Windows APIs.

**Comment by BurningPho3nix** [comment]
> I agree @vedantmgoyal2009.
> 
> One option would be to for Linux can do  query in WinGet repository, logically not upgrade nothing, only query and search.
> 
> If the idea is valid

This could actually help developers, who develop on Linux and package for Windows to have easier integrations with winget. 
Probably adding dry run for commands would help as well.

**Comment by beedell-roke** [comment]
> https://github.com/microsoft/winget-cli/issues/3889#issuecomment-3341404673

@BurningPho3nix, please update the original request to at least sincerely request query functionality, else you're wasting this opportunity (which you have sole discretion on).

**Comment by mcandre** [comment]
Porting winget to Linux, macOS, BSD's, HaikuOS, MINIX, z/OS, and the Solaris derivatives, would unlock a lot of applications currently unavailable to the world's developers.

**Comment by beedell-roke** [comment]
@mcandre, although macOS provides a more stable userland, and a significant userbase, and one could theorise that providing a BSD release would allow WinGet-registered software to be usable on embedded devices, no pragmatic use case for HaikuOS or MINIX exists.

Let us not reduce our chances of a sensible discussion by expanding the scope of the proposition.

**Comment by Murazaki** [comment]
There's actually good reasons to have winget working with wine, like being able to install Windows versions of some softwares easily, maybe integrating it with tools like winetricks, lutris,etc. and simplifying dev environment setup.
Saying it should not be done because Modern Windows API are not working with Wine is like saying we shall not eat because spoons are still in the drawer. That's actually what we are trying to change here.

**Comment by slackfan** [comment]
Just as a note which underlines the usefulness of the feature: There is https://github.com/Devolutions/pinget as a functional client for Linux.
